File: 10_14_2016_remove-nth-node-from-end-of-list/chang.java
/**
 * Definition for ListNode.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int val) {
 *         this.val = val;
 *         this.next = null;
 *     }
 * }
 */ 
public class Solution {
    /**
     * @param head: The first node of linked list.
     * @param n: An integer.
     * @return: The head of linked list.
     */
    ListNode removeNthFromEnd(ListNode head, int n) {
        // write your code here
        if (n <= 0 || head == null) {
            return null;
        }
        
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode predelete = dummy;
        //head = dummy;
        
        //move the first pointer to n
        for (int i = 0; i < n; i++) {
            head = head.next;
        }
        
        //move both pointers togerther
        while (head != null) {
            head = head.next;
            predelete = predelete.next;
        }
        
        predelete.next = predelete.next.next;
        
        return dummy.next;
        
        
    }
}

File: 11_19_2016-sort-list/chang.py
class Solution:
    """
    @param head: The first node of the linked list.
    @return: You should return the head of the sorted linked list,
                  using constant space complexity.
    """
    def sortList(self, head):
        # write your code here
        ans = []
        tmp = head
        while tmp != None:
            ans.append(tmp.val)
            tmp = tmp.next

        ans.sort()
        tmp = head
        for i in ans:
            tmp.val = i
            tmp = tmp.next
        return head
